### ISSUE TITLE
AAF read adapter should look for type Clip when it's finding a Marker's parent element from the Track

### ISSUE THREAD

**Issue by jchen9** [PixarAnimationStudios/OpenTimelineIO#1243]
In merge request https://github.com/PixarAnimationStudios/OpenTimelineIO/pull/1239, it left markers with the Track if `target_track.child_at_time(marker.marked_range.start_time)` returns something that does not have `markers` property, for example `Transition`.

This is less-than ideal. Instead, it should probably look for `Clip` and attach the marker to the clip found.

**Comment by reinecke** [comment]
From the TSC Meeting: Do we want to add some API so `child_at_time` can filter down so we avoid returning transitions? Do we ever want a transition returned by this method?

**Comment by jminor** [comment]
Yes, I think that is a good idea. The work-around #1239 has been merged for now, but your suggestion would be a cleaner fix.
Does it even make sense for child_at_time to ever return a Transition? Shouldn't a Transition always overlap a different child of the parent Composition?
